src/Tests.Microsoft.Security.Utilities.Core/WellKnownRegexPatternsTests.cs

```diff
@@ -26,22 +26,74 @@ public class WellKnownRegexPatternsTests
         };
 
+        [TestMethod]
+        public void WellKnownRegexPatterns_AllRuleIdsAndNamesAreUnique()
+        {            
+            using var assertionScope = new AssertionScope();
+
+            var patterns = GetAllPatterns();
+
+            HashSet<string> ruleIdsObserved = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            HashSet<string> ruleNamesObserved = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (RegexPattern pattern in patterns)
+            {
+                bool result = ruleIdsObserved.Contains(pattern.Id);
+                result.Should().BeFalse(because: $"Pattern '{pattern.GetType().Name}' should not share its Id with another rule: '{pattern.Id}'");
+
+                result = ruleNamesObserved.Contains(pattern.Name);
+                result.Should().BeFalse(because: $"Pattern '{pattern.GetType().Name}' should not share its Name with another rule: '{pattern.Name}'");
+
+                ruleIdsObserved.Add(pattern.Id);
+                ruleNamesObserved.Add(pattern.Name);
+            }
+        }
+
+        [TestMethod]
+        public void WellKnownRegexPatterns_MonikerRuleIdsAndNamesMatchDeclared()
+        {
+            using var assertionScope = new AssertionScope();
+
+            var patterns = GetAllPatterns();
+
+            foreach (RegexPattern pattern in patterns)
+            {
+                foreach (string example in pattern.GenerateTruePositiveExamples())
+                {
+                    var detection = pattern.GetDetections(example, generateCrossCompanyCorrelatingIds: false).FirstOrDefault();
+                    Assert.AreNotEqual(default, detection);
+
+                    string preciseMatch = example.Substring(detection.Start, detection.Length);
+
+                    var idAndName = pattern.GetMatchIdAndName(preciseMatch);
+
+                    Assert.AreEqual(pattern.Id, idAndName.Item1,
+                                    $"Pattern '{pattern.GetType().Name}' id did not match 'GetMatchIdAndName' result");
+
+                    Assert.AreEqual(pattern.Name, idAndName.Item2,
+                                    $"Pattern '{pattern.GetType().Name}' name did not match 'GetMatchIdAndName' result");
+                }
+            }
+        }
+
         [TestMethod]
         public void WellKnownRegexPatterns_GetMatchMonikerHardenedForOutOfOrderExecution()
         {
             using var assertionScope = new AssertionScope();
 
             var patterns = new List<RegexPattern>();
-
+            
             patterns.AddRange(WellKnownRegexPatterns.DataClassification);
             patterns.AddRange(WellKnownRegexPatterns.PreciselyClassifiedSecurityKeys);
-            patterns.AddRange(WellKnownRegexPatterns.UnclassifiedPotentialSecurityKeys);
+            patterns.AddRange(WellKnownRegexPatterns.UnclassifiedPotentialSecurityKeys);            
+
+            patterns.Add(new AzureFunctionIdentifiableKey());
```

Comment:
What is happening here that this rule needs to be added one-off. Should this be using GetAllPatterns()?